community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -99,6 +112,15 @@ def _find_dynamic_node_status() -> NodeStatus:
     return NodeStatus.unchanged  # don't touch dynamic nodes
 
 
+@lru_cache
+def get_keep_jobs():
```

Comment:
Why filter by `keep` state?

It should be a method on `Lookup`, otherwise cahce will be shared across multiple "Lookups", this would make for hard testing )

Reply:
Created a method in lookup to return all jobs details. Here we can filter by states as it was previously implemented.